cirq-core/cirq/ops/control_values.py

```diff
@@ -0,0 +1,170 @@
+# Copyright 2022 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import abc
+from typing import Union, Tuple, List, TYPE_CHECKING, Any, Dict
+from dataclasses import dataclass
+
+import itertools
+
+if TYPE_CHECKING:
+    import cirq
+
+
+@dataclass(frozen=True, eq=False)  # type: ignore
+class AbstractControlValues(abc.ABC):
+    """AbstractControlValues is an abstract immutable data class.
+
+    AbstractControlValues defines an API for control values and implements
+    functions common to all implementations (e.g.  comparison).
+    """
+
+    _internal_representation: Any
```

Comment:
`Any` is used only in the abstract class, concrete implementations use more strict annotations

Reply:
As discussed offline, it would be good to remove the internal implementation from the base class, and hence also remove the `_and_` method from base class for now. The latter can be added once the linked structure is introduced.

Reply:
moved the implementation to the concrete implementation class